FaceDetection/faceClassifer.py

```diff
@@ -0,0 +1,322 @@
+        for p, l in zip(preds, labels):
+            if p == l:
+                correct += 1
+        return correct / (len(preds))
+
+    def matplotlib_imshow(self, img, one_channel=False):
+        if one_channel:
+            img = img.mean(dim=0)
+        img = img / 2 + 0.5  # unnormalize
+        npimg = img.numpy()
+        if one_channel:
+            plt.imshow(npimg, cmap="Greys")
+        else:
+            plt.imshow(np.transpose(npimg, (1, 2, 0)))
+
+    def plot_results(self, metrics):
+        """ Simple plots of training and validation """
+        sns.lineplot(x=range(len(metrics['batch_train_loss'])), y=metrics['batch_train_loss'], color='red')
+        plt.xlabel('Batch num')
+        plt.ylabel('Train Loss')
+        plt.title('Training Batch Loss')
+        plt.show()
+
+        sns.lineplot(x=range(len(metrics['batch_val_loss'])), y=metrics['batch_val_loss'], color='orange')
+        plt.xlabel('Batch num')
+        plt.ylabel('Train Loss')
+        plt.title('Validation Batch Loss')
+        plt.show()
+
+        acc_df = pd.DataFrame({'val_acc_by_epoch': metrics['val_acc_by_epoch'],
+                               'train_acc_by_epoch':metrics['train_acc_by_epoch'],
+                               'epoch': range(len(metrics['train_acc_by_epoch']))})
+
+        sns.lineplot(data=acc_df,x='epoch', y='val_acc_by_epoch', color='blue')
+        sns.lineplot(data=acc_df,x='epoch', y='train_acc_by_epoch', color='orange')
+        plt.xlabel('Epoch')
+        plt.ylabel('Accuracy')
+        plt.title('Training and Validation Accuracy by Epoch')
+        plt.legend(['Validation', 'Training'])
+        plt.show()
+
+
+def test_accuracy_of_dataset(fc, dl, name):
+    """ Simple accuracy tester for face Classfication """
+    print('Accuracy of dataset --- ', name)
+    acc = []
+    for batch in dl:
+        images, labels = batch
+        inputs = images.to(fc.device)
+        labels = labels.to(fc.device)
+        preds = fc.predict(inputs)
+        acc.append(fc.get_accuracy(preds, labels))
+    print(np.mean(acc))
+
+def labelencode(label_encoder_output,X,y ,classes_to_drop:list):
+    """
+    creates a label encoder based on inserted classes to drop
+    Args:
+        label_encoder_output: output path to save label encoder
+        X: the image tensors
+        y: image labels
+        classes_to_drop: label classes not to be trained upon
+
+    Returns: X_transformed, y_transformed, label-encoder
+
+    """
+    y = [int(i) for i in y]
+    indexs_to_drop = [index for index, clss in enumerate(y) if clss in classes_to_drop]
+    X = [x for i, x in enumerate(X) if i not in indexs_to_drop]
+    y = [clss for i, clss in enumerate(y) if i not in indexs_to_drop]
+    assert len(X) == len(y)
+    le = preprocessing.LabelEncoder()
+    y_transformed = le.fit_transform(y)
+    pickle.dump(le, open(os.path.join(label_encoder_output, 'le.pkl'), 'wb'))
+    return X, y_transformed, le
+
+def main_train():
```

Comment:
Add comments above each line that describes what the line does, so we can understand which line should be removed according to what should be achieved in the run.
For example, in line 309: # load a pretrained model